shells/browser/shared/src/main.js

```diff
@@ -54,69 +63,85 @@ function createPanelIfReactLoaded() {
         // Otherwise the Store may miss important initial tree op codes.
         inject(chrome.runtime.getURL('build/backend.js'));
 
+        const viewElementSource = createViewElementSource(bridge, store);
+
+        const container = document.createElement('div');
+        const root = createRoot(container);
+
+        renderRootToPortal = ({ overrideTab, portalContainer }) => {
+          root.render(
+            createElement(DevTools, {
+              bridge,
+              browserName: getBrowserName(),
+              browserTheme: getBrowserTheme(),
+              overrideTab,
+              portalContainer,
+              showTabBar: false,
+              store,
+              viewElementSource,
+            })
+          );
+
+          const oldLinkTags = document.getElementsByTagName('link');
+          const newLinkTags = [];
+          for (let oldLinkTag of oldLinkTags) {
+            if (oldLinkTag.rel === 'stylesheet') {
+              const newLinkTag = document.createElement('link');
+              for (let attribute of oldLinkTag.attributes) {
+                newLinkTag.setAttribute(
+                  attribute.nodeName,
+                  attribute.nodeValue
+                );
+              }
+              newLinkTags.push(newLinkTag);
+            }
+          }
+
+          return newLinkTags;
+        };
+
         if (elementsPanel !== null) {
-          elementsPanel.injectBridgeAndStore(bridge, store);
-        }
-        if (profilerPanel !== null) {
-          profilerPanel.injectBridgeAndStore(bridge, store);
-        }
-        if (settingsPanel !== null) {
-          settingsPanel.injectBridgeAndStore(bridge, store);
+          elementsPanel.render(renderRootToPortal, 'elements');
         }
       }
 
       initBridgeAndStore();
 
-      chrome.devtools.panels.create(
-        '⚛ Elements',
-        '',
-        'elements.html',
-        panel => {
-          panel.onShown.addListener(panel => {
-            if (elementsPanel === null) {
-              panel.injectBridgeAndStore(bridge, store);
-            }
+      chrome.devtools.panels.create('⚛ Elements', '', 'panel.html', panel => {
+        panel.onShown.addListener(panel => {
+          elementsPanel = panel;
 
-            elementsPanel = panel;
+          if (renderRootToPortal !== null) {
+            elementsPanel.render(renderRootToPortal, 'elements');
+          }
 
-            // TODO: When the user switches to the panel, check for an Elements tab selection.
-          });
-          panel.onHidden.addListener(() => {
-            // TODO: Stop highlighting and stuff.
-          });
-        }
-      );
+          // TODO: When the user switches to the panel, check for an Elements tab selection.
+        });
+        panel.onHidden.addListener(() => {
+          // TODO: Stop highlighting and stuff.
+        });
+      });
 
       // TODO (profiling) Is there a way to detect profiling support and conditionally register this panel?
```

Comment:
From a user's perspective it may be unclear why the panel is missing and what to do to enable profiling. Having the panel regardless gives an opportunity to explain this to the user on that panel.

Reply:
This matches the current DevTools behavior.

Reply:
> This matches the current DevTools behavior.

👍